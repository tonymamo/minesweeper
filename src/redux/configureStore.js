import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { apiMiddleware } from 'redux-api-middleware';

import rootReducer from './reducers/index';
import logger from './logger';

let middleware = [
    thunk,
    apiMiddleware,
    logger,
];

export const createStoreWithMiddleware = compose(
    applyMiddleware(...middleware),
    typeof window === 'object' && typeof window.devToolsExtension !== 'undefined' ? window.devToolsExtension() : f => f
)(createStore);

export default function configureStore(initialState) {
    const store = createStoreWithMiddleware(rootReducer, initialState);

    if ( module.hot ) {
        module.hot.accept('./reducers', () => {
            const nextReducer = require('./reducers').default;
            store.replaceReducer(nextReducer);
        });
    }

    return store;
}
