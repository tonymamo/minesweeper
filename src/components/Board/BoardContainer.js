import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import * as boardActionCreators from '../../redux/actions/board';
import * as appActionCreators from '../../redux/actions/app';
import Board from './Board';

class BoardContainer extends Component {
    render() {
        return <Board {...this.props}/>;
    }
}

function mapStateToProps(state) {
    return {
        board:          state.get('board'), 
        totalOpened:    state.get('totalOpened'), 
        minesLeft:      state.get('minesLeft'),
        completed:      state.get('completed'),
        gameOver:       state.get('gameOver')
    };
}

function mapDispatchToProps(dispatch) {
    return {
        boardActions: bindActionCreators(boardActionCreators, dispatch),
        appActions: bindActionCreators(appActionCreators, dispatch)
    };
}

BoardContainer.propTypes = {
    boardActions: PropTypes.object,
    appActions: PropTypes.object
};

export default connect(mapStateToProps, mapDispatchToProps)(BoardContainer);
