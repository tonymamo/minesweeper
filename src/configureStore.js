import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { apiMiddleware } from 'redux-api-middleware';

import rootReducer from './redux/index';
import logger from './redux/logger';

let middleware = [
    thunk,
    apiMiddleware,
    logger,
];

export const createStoreWithMiddleware = compose(
    applyMiddleware(...middleware),
    typeof window === 'object' && typeof window.devToolsExtension !== 'undefined' ? window.devToolsExtension() : f => f
)(createStore);

export default function configureStore(initialState) {
    const store = createStoreWithMiddleware(rootReducer, initialState);

    if ( module.hot ) {
        module.hot.accept('./redux', () => {
            const nextReducer = require('./redux').default;
            store.replaceReducer(nextReducer);
        });
    }

    return store;
}
