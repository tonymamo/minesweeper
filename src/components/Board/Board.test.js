import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import configureStore from '../../redux/configureStore';
import BoardContainer from './BoardContainer';

const store = configureStore();

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Provider store={store}><BoardContainer /></Provider>, div);
});

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}><BoardContainer /></Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
