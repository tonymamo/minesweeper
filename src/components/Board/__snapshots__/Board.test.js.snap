// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="board"
>
  <div
    className="board__header"
  >
    <div
      className="counter"
    >
      <span
        className="pill"
      >
        10
      </span>
       
      <span
        className="board__subheading"
      >
        Mines Remaining
      </span>
    </div>
    <div
      className="counter"
    >
      <span
        className="pill"
      >
        0
      </span>
       
      <span
        className="board__subheading"
      >
        Time Elapsed
      </span>
    </div>
  </div>
  <p
    className="align--center"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      Restart
    </button>
  </p>
  <div
    onContextMenu={[Function]}
  />
</div>
`;
